src/nccl_ofi_rdma.c

```diff
@@ -1598,11 +1605,11 @@ static ssize_t receive_progress(nccl_net_ofi_rdma_req_t *req, bool add_to_pendin
 		/* Extract ep */
 		nccl_net_ofi_rdma_ep_t *ep = (nccl_net_ofi_rdma_ep_t *)r_comm->base.base.ep;
 		/* Place in pending requests queue for next try */
-		int ret = nccl_ofi_deque_insert_back(ep->pending_reqs_queue,
-						     &req->pending_reqs_elem);
-		if (ret != 0) {
-			NCCL_OFI_WARN("Failed to nccl_ofi_deque_insert_back: %d", ret);
-			return ret;
+		rc = nccl_ofi_deque_insert_back(ep->pending_reqs_queue,
```

Comment:
The return type of this functio and the subsequent return code handling all expects zero or negative return types stored in ssize_t (which probably ends up being projected in range [-1, 2^15-1] or so).

It's debatable whether that is how one wants to treat initially negative error codes. That being said, in my opinion the current bugfix is in line with the paradigma we currently follow and in my tests indeed fixes the bug. Also, just setting `rc = 0` was not what I wanted to do since I also want to report the error from `nccl_ofi_deque_insert_back()` back in case `receive_progress()` errors due to an error in `nccl_ofi_deque_insert_back()`. By doing that, we get a correct root cause error reporting ( even including nagative values as soon as we refactor `ssize_t` to `int`, which is in the pipeline as far as I understood).

What do you think?

Reply:
I think this entire section should be unchanged, except for adding `rc = 0;` on line 1614.